laurent22/joplin
# Chore: Migrate file system sync to TypeScript

# Summary

This pull request migrates `packages/lib/SyncTargetFilesystem.js` and `packages/lib/file-api-driver-local.js` to TypeScript.



# Testing

1. Replace `setSyncTargetName('memory');` with `setSyncTargetName('filesystem');` and run `yarn test` from `packages/lib`
    - One test failed with the error
    ```
    Error: Error: ENOENT: no such file or directory, open '/home/builder/Documents/joplin/packages/lib/testing/../../app-cli/tests/sync/TkZLOsbb1hR2b9rT63LKIO/.resource/bf701ff958fd4583b18c0e1eb19d698e'. Path: /home/builder/Documents/joplin/packages/lib/testing/../../app-cli/tests/sync/TkZLOsbb1hR2b9rT63LKIO/.resource/bf701ff958fd4583b18c0e1eb19d698e. Path: /home/builder/Documents/joplin/packages/lib/testing/../../app-cli/tests/sync/TkZLOsbb1hR2b9rT63LKIO/.resource/bf701ff958fd4583b18c0e1eb19d698e
    ```
    This was fixed by adjusting the test to `mkdir('.resources')` before attempting to write a file to it.
2. Start the desktop app and configure file system sync with an empty directory. Verify that files are written to it without errors.
    - I tested with a single note with a single resource.
3. Create a new profile and sync with that directory.
    - The resource and note were synced successfully.
4. On mobile, create a new profile with a single note and notebook.
    - Tested on an Android 10 emulator.
5. Add a drawing to the note.
6. Set up file system sync with a new, empty directory.
7. Create and switch to a new profile.
8. Synchronize with the just-created directory.
9. Verify that the resource, note, and notebook are synced.
